gtk3/sample/misc/textbuffer_serialize.rb

```diff
@@ -20,100 +20,106 @@
   
   dialog.filename = File.expand_path(file_name)
   if dialog.run == Gtk::ResponseType::ACCEPT
     file_name = dialog.filename
     File.open(file_name, "rb") {|io|
       buffer.delete(buffer.start_iter, buffer.end_iter)
-      buffer.deserialize(buffer, format, buffer.start_iter, io.read)
+      buffer.deserialize(buffer, deserialize_format, buffer.start_iter, io.read)
     }
   end
   dialog.destroy
 end
 
-toolbar.append(Gtk::Stock::SAVE, "Serialize to a file") do
-  dialog = Gtk::FileChooserDialog.new("Serialize from a file",
-                                      window,
-                                      Gtk::FileChooserAction::SAVE,
-                                      nil,
-                                      [Gtk::Stock::CANCEL, Gtk::ResponseType::CANCEL],
-                                      [Gtk::Stock::OPEN, Gtk::ResponseType::ACCEPT])
+button_save = Gtk::ToolButton.new(:icon_widget => nil, :label => "Serialize to a file",:stock_id => Gtk::Stock::SAVE)
+button_save.signal_connect "clicked" do 
+  dialog = Gtk::FileChooserDialog.new(:title => "Serialize to a file",
+                                      :parent => window,
+                                      :action => :save, #or Gtk::FileChooserAction::SAVE,
+                                      :buttons => [
+                                        [Gtk::Stock::CANCEL, Gtk::ResponseType::CANCEL],
+                                        [Gtk::Stock::OPEN, Gtk::ResponseType::ACCEPT]
+                                      ])
   dialog.current_name = file_name
   if dialog.run == Gtk::ResponseType::ACCEPT
     file_name = dialog.filename
     File.open(file_name, "wb") {|io|
-      io.write(buffer.serialize(buffer, format, buffer.start_iter, buffer.end_iter))
+      data = buffer.serialize(buffer, format, buffer.start_iter, buffer.end_iter)
+      io.write(data.pack("C*"))
     }
   end
   dialog.destroy
 end
 
-toolbar.append(Gtk::Stock::CLEAR, "Clear all") do
-  buffer.delete(buffer.start_iter, buffer.end_iter)
-end
-toolbar.append_space
-toolbar.append(Gtk::Stock::SELECT_COLOR, "Color the region") do
-  dialog = Gtk::ColorSelectionDialog.new("Color the region")
+button_clear = Gtk::ToolButton.new(:icon_widget => nil, :label => "Clear all",:stock_id => Gtk::Stock::CLEAR)
+button_clear.signal_connect("clicked") { buffer.delete(buffer.start_iter, buffer.end_iter) }
+toolbar.insert(button_clear, 0)
+
+button_color = Gtk::ToolButton.new(:icon_widget => nil, :label => "Color the region",:stock_id => Gtk::Stock::SELECT_COLOR)
+button_color.signal_connect "clicked" do 
+  dialog = Gtk::ColorChooserDialog.new(:title => "Color the region", :parent => window)
   if dialog.run == Gtk::ResponseType::OK
-    bounds = buffer.selection_bounds
-    color = dialog.colorsel.current_color
-    tag_name = color.to_a.inspect
+    bounds = buffer.selection_bounds # returns an array [test_selected(t/f)? ,Gtk::TextIter, Gtk::TextIter]
```

Comment:
I improved the API at master. This returns `[Gtk::TextIter, Gtk::TextIter]` at master.